autoprofile.py

```diff
@@ -33,15 +35,91 @@ def __init__(self):
         self.name = _("Automatic Profile")
         self.bio_enabled = False
         self.name_enabled = False
+        self.pfp_enabled = False
+        self.pfp_remove_latest = None
         self.raw_bio = None
         self.raw_name = None
+        self.pfp_degree = 0
 
     async def client_ready(self, client, db):
         self.client = client
 
+    async def autopfpcmd(self, message):
+        """Rotates your profile picture every n seconds with x degrees, usage:
+           (prefix)autopfp <timeout> <degrees> <remove previous (last pfp)>
+
+           Timeout - seconds
+           Degrees - 60, -10, etc
+           Remove last pfp - true or false, boolean, case insensitive"""
+
+        msg = utils.get_args(message)
+        if len(msg) != 3:
+            return await message.edit(_("<b>Autopfp requires three args. See the help for syntax.</b>"))
+
+        try:
+            timeout_autopfp = int(msg[0])
+        except ValueError as e:
+            logger.warning(str(e))
+            return await message.edit(_("<b>Wrong time.</b>"))
+
+        try:
+            degrees = int(msg[1])
+        except ValueError as e:
+            logger.warning(str(e))
+            return await message.edit(_("<b>Wrong degrees value.</b>"))
+
+        if msg[2].lower() == 'true':
+            delete_previous = True
+        elif msg[2].lower() == 'false':
+            delete_previous = False
+        else:
+            return await message.edit(_("<b>Please pass true or false for previous pfp removing.</b>"))
+
+        client = self.client
+        me = await client.get_me()
+        pfp = await client.download_profile_photo(me, file=bytes)
+        raw_pfp = Image.open(BytesIO(pfp))
+
+        self.pfp_remove_latest = delete_previous
+        self.pfp_enabled = True
+        await message.edit("<b>Successfully enabled autopfp.</b>")
+
+        while self.pfp_enabled:
+            self.pfp_degree += degrees
+            rotated = raw_pfp.rotate(self.pfp_degree)
+            buf = BytesIO()
+            rotated.save(buf, format='JPEG')
+            bytes_img = buf.getvalue()
+
+            if self.pfp_remove_latest:
+                await client(functions.photos.DeletePhotosRequest(
+                    await client.get_profile_photos(await client.get_me(), limit=1)
+                ))
```

Comment:
Doesn't that need [0]?

Reply:
nope it doesnt, if do that it errors with `TypeError: 'Photo' object is not iterable`

Reply:
Oh well I guess telethon is doing some internal shit